app/models/license.rb

```diff
@@ -0,0 +1,84 @@
+#-- copyright
+# OpenProject is a project management system.
+# Copyright (C) 2012-2017 the OpenProject Foundation (OPF)
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2017 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See doc/COPYRIGHT.rdoc for more details.
+#++
+class License < ActiveRecord::Base
+  class << self
+    def current
+      set_current_license unless defined?(@@current_license)
+      @@current_license
```

Comment:
@oliverguenther I believe that using class variables will "cache" the value even across requests, am I right? If so, how do we make sure that a new license can take effect, or that a Iicense can be deleted? For my understanding we need a request wide/limited caching, such as:

```ruby
  class << self
    def current
      if RequestStore.store[:current_license].present?
        return RequestStore.store[:current_license]
      else
        license = License.order('created_at DESC').first
        if license.nil?
          return nil
        else
          RequestStore.store[:current_license] = license
        end
      end
    end
  end
```

Fortunately `RequestStore` is already in the Gemfile so it does not create new dependencies.